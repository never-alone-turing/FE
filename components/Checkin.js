import React from "react";
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";

class Checkin extends React.Component {
  constructor() {
    super()
    this.state = {
      currentStyle: styles.mainBtn
    }
  }

  changeDecider = () => {
    if (this.state.currentStyle === styles.mainBtn) {
      this.state.currentStyle = styles2.mainBtn
    } else {
      this.state.currentStyle = styles.mainBtn
    }
    this.forceUpdate()
  }

  render() {
    return (
      <TouchableOpacity style={this.state.currentStyle} onLongPress={() => this.changeDecider()}>
        <View>
          <Text>Touch me</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  mainBtn: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#69c986',
    height: 300,
    width: 300,
    borderRadius: 300 / 2
  }
})

const styles2 = StyleSheet.create({
  mainBtn: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#f5a3af',
    height: 300,
    width: 300,
    borderRadius: 300 / 2
  }
})

export default Checkin;