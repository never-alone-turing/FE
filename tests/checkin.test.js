import React from 'react'
import {create} from 'react-test-renderer'
import Checkin from '../components/Checkin'
import renderer from 'react-test-renderer'


  describe("<Checkin />", () => {

    it('renders correctly', () => {
      const tree = renderer.create(<Checkin />).toJSON();
    expect(tree).toMatchSnapshot()
    })

  })
