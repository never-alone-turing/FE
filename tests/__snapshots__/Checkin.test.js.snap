// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkin should match the snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#70cfff",
      "borderColor": "#5AA6CC",
      "borderRadius": 150,
      "borderWidth": 20,
      "display": "flex",
      "height": 450,
      "justifyContent": "center",
      "marginBottom": 80,
      "opacity": 1,
      "width": 450,
    }
  }
  testID="checkin-1"
>
  <View>
    <Text
      style={
        Object {
          "fontSize": 50,
          "textAlign": "center",
        }
      }
    >
      Not available to check in yet
    </Text>
  </View>
</View>
`;
